lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+""" API implementation for course-oriented interactions. """
+
+import logging
+
+from django.core.urlresolvers import reverse
+from django.http import Http404
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework import generics
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+
+from courseware.courses import course_image_url
+from course_api.courseware_access import get_course, get_course_child, get_course_descriptor
+from openedx.core.lib.api.views import APIViewWithPermissions, PaginatedListAPIViewWithPermissions, PermissionMixin
+from course_api.v0.serializers import CourseSerializer, GradedContentSerializer, GradingPolicySerializer
+
+
+log = logging.getLogger(__name__)
+
+
+def _get_content_children(content):
+    """
+    Parses the provided content object looking for children.
+    """
+    children = []
+    if content.has_children:
+        child_content = content.get_children()
+        for child in child_content:
+            children.append(child)
+    return children
+
+
+def _serialize_content(request, course_key, content_descriptor):
+    """
+    Loads the specified content object into the response dict
+    This should probably evolve to use DRF serializers
+    """
+
+    data = {}
+
+    if hasattr(content_descriptor, 'display_name'):
+        data['name'] = content_descriptor.display_name
+
+    if hasattr(content_descriptor, 'due'):
+        data['due'] = content_descriptor.due
+
+    data['start'] = getattr(content_descriptor, 'start', None)
+    data['end'] = getattr(content_descriptor, 'end', None)
+
+    data['category'] = content_descriptor.location.category
+
+    # Some things we only do if the content object is a course
+    if hasattr(content_descriptor, 'category') and content_descriptor.category == 'course':
```

Comment:
This would be more future-proof as `if content_descriptor.scope_ids.block_type == 'course'`